token-lending/program/src/processor.rs

```diff
@@ -1626,12 +1631,15 @@ fn process_flash_loan(program_id: &Pubkey, amount: u64, accounts: &[AccountInfo]
     let after_liquidity_supply_token_account =
         Account::unpack_from_slice(&source_liquidity_info.try_borrow_data()?)?;
 
-    let expected_amount = reserve.liquidity.available_amount + origination_fee;
-    if after_liquidity_supply_token_account.amount < expected_amount {
+    let total_expected_amount = before_liquidity_supply_token_account
+        .amount
+        .checked_add(origination_fee)
+        .ok_or(LendingError::MathOverflow)?;
```

Comment:
This can be calculated before the loan is made. You can also just get the value you need from the account, e.g.

```rust
let balance_before_loan = Account::unpack_from_slice(
    &source_liquidity_info.try_borrow_data()?
)?.amount;
let expected_balance_after_loan = balance_before_loan.checked_add(origination_fee)
    .ok_or(LendingError::MathOverflow)?;
```

Reply:
Fixed.